Description: Add Make.config to compile debug versions.
Author: Tobias Grimm <vdr@e-tobi.net>
Re-Authored: Lars Hanisch <dvb@flensrocker.de>

--- /dev/null
+++ b/Make.config
@@ -0,0 +1,95 @@
+#
+# User defined Makefile options for the Video Disk Recorder
+#
+# Copy this file to 'Make.config' and change the parameters as necessary.
+#
+# See the main source file 'vdr.c' for copyright information and
+# how to reach the author.
+#
+# $Id: Make.config.template 2.18 2013/02/08 10:31:38 kls Exp $
+
+### The C compiler and options:
+
+CC       = gcc
+CFLAGS   = -g -Wall
+
+CXX      = g++
+CXXFLAGS = -g -Wall -Werror=overloaded-virtual -Wno-parentheses
+
+ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
+    CFLAGS += -O0
+    CXXFLAGS += -O0
+else
+    CFLAGS += -O3
+    CXXFLAGS += -O3
+endif
+
+ifdef VDRDEBUG
+    # Required for special debug build of VDR, where plugins are
+    # not unloaded at runtime to give valgrind a chance to detect
+    # memory leaks in the plugin libs
+    DEFINES += -DVDRDEBUG
+endif
+
+BIDI = 1
+
+# Use 'make M32=1 ...' to build a 32-bit version of VDR on a 64-bit machine:
+ifdef M32
+CFLAGS   += -m32
+CXXFLAGS += -m32
+endif
+
+### The directory environment:
+
+# Default directories (adjust as necessary or desired):
+
+PREFIX    = /usr
+#BINDIR    = $(PREFIX)/bin
+#INCDIR    = $(PREFIX)/include
+LIBDIR    = $(PREFIX)/lib/vdr/plugins
+#LOCDIR    = $(PREFIX)/share/locale
+#MANDIR    = $(PREFIX)/share/man
+#PCDIR     = $(PREFIX)/lib/pkgconfig
+#RESDIR    = $(PREFIX)/share/vdr
+#DVBDIR    = /usr/src/v4l-dvb/linux/include/uapi
+
+VIDEODIR  = /srv/vdr/video.00
+#CONFDIR   = /var/lib/vdr
+#CACHEDIR  = /var/cache/vdr
+
+# Overrides for preset/legacy configurations:
+
+# Use 'make LCLBLD=1' to build locale and plugin files under the source directory:
+ifdef LCLBLD
+LOCDIR       = $(CWD)/locale
+PLUGINDIR    = $(CWD)/PLUGINS
+ifndef PLUGIN # don't overwrite for plugins with old makefiles
+LIBDIR       = $(PLUGINDIR)/lib
+endif
+HDRDIR       = $(CWD)/include
+endif
+
+# Use 'make ONEDIR=1' to have all data in one single directory:
+ifdef ONEDIR
+VIDEODIR     = /video
+CACHEDIR     = $(VIDEODIR)
+CONFDIR      = $(VIDEODIR)
+RESDIR       = $(VIDEODIR)
+endif
+
+# Use this if you want to have a central place where you configure compile time
+# parameters for plugins:
+#PLGCFG = $(CONFDIR)/plugins.conf
+
+### The remote control:
+
+LIRC_DEVICE = /var/run/lirc/lircd
+
+### Define if you want vdr to not run as root:
+#VDR_USER = vdr
+
+### Fallback for plugins with old makefiles:
+ifdef PLUGIN
+CFLAGS   += -fPIC
+CXXFLAGS += -fPIC
+endif
